src/native/libs/System.Globalization.Native/CMakeLists.txt

```diff
@@ -1,19 +1,23 @@
 project(System.Globalization.Native C)
 
-if(CLR_CMAKE_TARGET_UNIX)
+if(CLR_CMAKE_TARGET_UNIX OR CLR_CMAKE_TARGET_WASI)
     add_compile_options(-Wno-switch-enum)
     add_compile_options(-Wno-covered-switch-default)
 
     # Workaround for warnings produced by ICU headers
     add_compile_options(-Wno-reserved-id-macro)
     add_compile_options(-Wno-documentation)
     add_compile_options(-Wno-documentation-unknown-command)
+    add_compile_options(-Wno-reserved-identifier)
 
     # Workaround for https://unicode-org.atlassian.net/browse/ICU-20601
     add_compile_options(-Wno-extra-semi-stmt)
     add_compile_options(-Wno-unknown-warning-option)
 
-    if (NOT CLR_CMAKE_TARGET_ANDROID)
+    # pal_icushim_static.c(41,24): error G026380BD: unused function 'icu_trace_data'
+    add_compile_options(-Wno-unused-function)
```

Comment:
we should instead wrap the `icu_trace_data` function with `#if defined(ICU_TRACING)`